lib/endpoints/class-wp-json-terms-controller.php

```diff
@@ -368,4 +379,24 @@ protected function check_valid_taxonomy( $taxonomy ) {
 		}
 	}
 
+	/**
+	 * Check that the taxonomy is valid for a post.
+	 *
+	 * @param string  $taxonomy
+	 * @param integer $post_id
+	 * @return string|WP_Error
+	 */
+	protected function check_valid_taxonomy_for_post( $taxonomy, $post_id ) {
+		$post = get_post( $post_id );
+		if ( empty( $post->ID ) ) {
+			return new WP_Error( 'json_post_invalid_id', __( 'Invalid post ID.' ), array( 'status' => 404 ) );
+		}
+
+		$valid_taxonomies = get_object_taxonomies( $post->post_type );
+		if ( in_array( $taxonomy, $valid_taxonomies ) ) {
+			return $taxonomy;
```

Comment:
Why isn't this just returning `bool`? I don't understand why you'd need `$taxonomy` returned when you're passing it in the first place.